# Aluna: Gabriela Guerra
Entrega da Semana 1

Materiais:
*   O que é eng de dados
https://www.databricks.com/br/blog/what-is-data-engineering
*   Tutorial Pandas
https://colab.research.google.com/github/Michell-Piazza/Pandas-Tutorial/blob/main/Tutorial_Pandas.ipynb/
*   Tutorial Git Hub
https://docs.github.com/en/get-started/start-your-journey/hello-world

Dataset txt. usado:
associação entre piRNA e doenças
http://www.biomedical-web.com/pirpheno/Download.jsp



In [ ]:
import pandas as pd
import numpy as np

In [ ]:
from google.colab import drive

drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/PANDA/prática 1/piRNA_disease_association.txt", sep="\t") #read_csv(sep="\t") é arquivo TXT que é separado por TAB

In [ ]:
df.head()
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9057 entries, 0 to 9056
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Disease parent     9057 non-null   object 
 1   Association id     9057 non-null   object 
 2   Disease name       9057 non-null   object 
 3   Disease ontology   9057 non-null   object 
 4   piRNA              9057 non-null   object 
 5   Association score  9057 non-null   float64
 6   Correlation        9057 non-null   object 
dtypes: float64(1), object(6)
memory usage: 495.4+ KB


,Association score
count,9057.000000
mean,0.078884
std,0.105418
min,0.000000
25%,0.050000
50%,0.050000
75%,0.050000
max,1.000000


In [ ]:
df.columns
df.isnull().sum()

,0
Disease parent,0
Association id,0
Disease name,0
Disease ontology,0
piRNA,0
Association score,0
Correlation,0


RESUMO

*   df.head()
mostra as primeiras linhas da tabela
como os dados estão organizados
*   df.info()
resumo geral do dataset
ex. 9057 entries, 7 columns, object: texto, float64: número com decimal
*   df.describe()
resumo estatístico dos números
*   df.columns
nome das colunas
*   df.isnull()
verifica onde tem valores vazios
true: vazio
false: preenchido
*   df.isnull().sum()
conta quantos valores vazios tem em cada coluna
ex. tudo deu 0 = dataset está completo (sem dados faltando)







In [ ]:
df.groupby("piRNA")["Disease name"].nunique().sort_values(ascending=False) #quantas doenças cada piRNA tem?

,Disease name
piRNA,
piR-hsa-1282,92
piR-hsa-20266,71
piR-hsa-1077,69
piR-hsa-1963,68
piR-hsa-26570,62
...,...
piR-hsa-1207,8
piR-hsa-6433,8
piR-hsa-2107,8


In [ ]:
df.groupby("Disease name")["piRNA"].nunique().sort_values(ascending=False)  #quantos piRNAs diferentes estão associados a doença?

,piRNA
Disease name,
Disease by anatomical system,465
Cell proliferation disorder,356
Neoplastic disease or syndrome,356
Neoplasm,356
Cancer,344
...,...
Osteoarthritis,1
Squamous cell lung carcinoma,1
Rheumatic disease,1


In [ ]:
df["Association score"].value_counts() #distribuição do score

,count
Association score,
0.05,7465
0.10,822
0.20,303
0.40,130
0.80,76
0.00,69
0.45,61
0.15,45
0.60,28


Computation of association score: http://www.biomedical-web.com/pirpheno/help.jsp#score

O association score é calculado integrando três fontes de informação: evidências diretas de associações já reportadas entre piRNAs e doenças, similaridade entre doenças (permitindo inferir associações a partir de doenças relacionadas) e similaridade entre piRNAs (permitindo transferir informação entre moléculas semelhantes), resultando em um valor que representa o nível de confiança na associação.

ex. 0.05 - menor suporte/menos evidência
ex. 0.6 - maior suporte/maior evidência

In [ ]:
df["Disease name"].value_counts()  #frequência das doenças

,count
Disease name,
Disease by anatomical system,465
Neoplasm,356
Neoplastic disease or syndrome,356
Cell proliferation disorder,356
Cancer,344
...,...
Bone inflammation disease,1
Epithelial tumor of colon,1
Bone marrow neoplasm,1


In [ ]:
df.groupby("piRNA").size().value_counts() #associação piRNA com mais de 1 doença

,count
16,182
11,89
23,40
28,31
8,23
21,22
33,13
14,8
32,7
37,7


ex.
de 182 piRNAs diferentes, cada um deles está associada a 16 doenças diferentes

In [ ]:
df[df["Association score"] > 0.05] #filtro associações fortes

,Disease parent,Association id,Disease name,Disease ontology,piRNA,Association score,Correlation
3,Male reproductive system disease,PIRP00000004,Teratozoospermia,EFO_0002625,piR-hsa-29218,0.1,Negative
8,Malignant urinary system neoplasm,PIRP00000009,Kidney cancer,MONDO_0002367,piR-hsa-12178,0.2,Positive
9,Intestinal neoplasm,PIRP00000010,Intestinal cancer,EFO_0007330,piR-hsa-1282,0.6,Positive
10,Male infertility with spermatogenesis disorder...,PIRP00000011,Male infertility with azoospermia or oligozoos...,Orphanet_399805,piR-hsa-29218,0.1,Negative
16,Germ cell tumor,PIRP00000017,Malignant germ cell tumor,EFO_1000352,piR-hsa-10476,0.1,Negative
...,...,...,...,...,...,...,...
9019,Reproductive system disease,PIRP00009020,Infertility,EFO_0000545,piR-hsa-2107,0.2,Negative
9021,Disease,PIRP00009022,Disease by anatomical system,MONDO_0021199,piR-hsa-7247,0.1,Negative
9025,Thoracic disease,PIRP00009026,Bronchial disease,EFO_1002018,piR-hsa-26606,0.1,Negative
9039,Epithelial neoplasm,PIRP00009040,Squamous cell neoplasm,MONDO_0002532,piR-hsa-23209,0.1,Positive
